### ISSUE TITLE
GPL-3 license?

### ISSUE THREAD

**Issue by tshort** [mkborregaard/MakieThemes.jl#3]
Seems like a conflict with the rest of the plotting ecosystem. Why is it needed anyway? I understand that GPL is viral, but can the look of a plot be copyrightable?

**Comment by tshort** [comment]
Here are some non-GPL themes from the Python world:

https://tonysyu.github.io/raw_content/matplotlib-style-gallery/gallery.html

**Comment by mkborregaard** [comment]
> Seems like a conflict with the rest of the plotting ecosystem

It is different from the rest of the plotting ecosystem, but there is no conflict. IMHO there is absolutely no reason to be concerned about the GPL license in a self-contained place like this. I also sometimes feel that the MIT-crusade in the Julia ecosystem takes on an unnecessarily strong character. (All I do myself is MIT but that's because I think it's the right license for it, not out of fear of the GPL).

The reason for having this GPL-3 was that I wanted to implement existing themes from amongst others the R ecosystem, and in R everything is GPL-3. Also, making this GPL-3 globally should contradict no other license. 

I think the matplotlib themes are nice, and indeed a planned project is to make the matplotlib themes available in this package as well (that's why GGThemr is already organized as a submodule).

**Comment by tshort** [comment]
Because of the infectious nature of the GPL, it's harder to get permission at my company to use anything with a GPL license that might become published. Based on that, I avoid using GPL packages.

If there's a way to separate out the matplotlib themes to avoid GPL on that, it'd help.

**Comment by mkborregaard** [comment]
Hm, that's a relevant concern. But does that apply to _use_ too, or only to depending on the package? My reasoning is that no package would want to depend on this, but that it would just be loaded directly by the end user, so there shouldn't be a problem (by the way depending on a package that is GPL should not make your package GPL as long as it's nice and self-contained, but some people/companies do tend to avoid it just to make sure).

In principle I could license the individual submodules - it should still be possible to have the overall module MIT I guess but I'm a bit uncertain here.

**Comment by tshort** [comment]
For everyday use, it's not a problem for me. When it becomes a problem is if I want to include a theme into code delivered as a bundle where the bundle could be a program or package. Then, the GPL infects the whole bundle, and it complicates delivery even if it's licensed as open source. For me, that use case is still probably a year or two off. I'm hoping Julia's compiling capability improves over that time period and it also becomes easier to distribute GPL-free code. It's enough of an incentive for me to try not to get too attached to GPL code.

**Comment by mkborregaard** [comment]
I guess that should be fine for this, as it only changes appearance of plots but otherwise no other functionality. Still, I'd happily consider any alternative solution that is not "only use MIT licensed themes"

**Comment by tshort** [comment]
I want to clarify one part. As long as I don't look at the GPL code, I can make an MIT-licensed theme that looks like R's ggplot, right? (They did that in matplotlib.)

**Comment by mkborregaard** [comment]
I don't think so on principle. There is no clause in the GPL license talking about looking at code (similarly, the notion that you can not look at GPL code is an extreme interpretation widespread among proponents of MIT) . They talk about derived work. The problem being that that is har to define. Matlab will also sue you if you use their parula colour scale even though you got it just from looking at the colours.

**Comment by mkborregaard** [comment]
But mostly noone will prosecute you if you do it. People don't GPL-license their code in order to prosecute you or anyone. People GPL-license code because they don't want some company to make a lot of money out of closed-sourcing a version of code they have produced for free out of their love for open source.
As such, I also find "cheating" on the license by copying the theme without looking at the code unethical, even if it legally is a border case.

**Comment by tshort** [comment]
Another option is to ask the ggthemr folks if they were okay to license this derivative with an MIT license (there are just two of them).

**Comment by tshort** [comment]
Another licensing issue is that themes from the [gggthemes](https://yutannihilation.github.io/allYourFigureAreBelongToUs/ggthemes/) package cannot be included here because it is GPL-2 licensed.

**Comment by mkborregaard** [comment]
I'll try and ask them, that would be a nice and clean solution for this case at least.

**Comment by mkborregaard** [comment]
https://github.com/cttobin/ggthemr/issues/30
